community/security/src/main/java/org/neo4j/server/security/auth/RateLimitedAuthenticationStrategy.java

```diff
@@ -94,7 +94,7 @@ public AuthenticationResult authenticate( User user, String password)
     {
         AuthenticationMetadata authMetadata = authMetadataFor( user.name() );
 
-        if ( !isAuthenticationPermitted( user.name() ) )
+        if ( !authMetadata.authenticationPermitted() )
```

Comment:
Should remove the unused `isAuthenticationPermitted` and `updateWithAuthenticationResult`